task/sast-coverity-check/0.1/sast-coverity-check.yaml

```diff
@@ -0,0 +1,264 @@
+        set -eo pipefail
+        . /usr/local/share/konflux-test/utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+        
+        echo 'Starting Coverity buildless scan'
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)
+        COVERITY_DIR=/var/tmp/coverity/idir
+        COVERITY_RESULTS_FILE=$(workspaces.workspace.path)/hacbs/$(context.task.name)/coverity-buildless-results.json
+        COV_LICENSE_PATH=/etc/secrets/cov/cov-license
+        # Installing Coverity license
+        cp "$COV_LICENSE_PATH" /opt/coverity/bin/license.dat
+        if test -z "$(find /opt -name 'license.dat')"; then
+          echo "No license file for Coverity was detected. Exiting..."
+          exit 1
+        fi
+        
+        # Installation of Red Hat certificates for cloning Red Hat internal repositories
+        ca_bundle=/mnt/trusted-ca/ca-bundle.crt
+        if [ -f "$ca_bundle" ]; then
+          echo "INFO: Using mounted CA bundle: $ca_bundle"
+          cp -vf $ca_bundle /etc/pki/ca-trust/source/anchors
+          update-ca-trust
+        fi
+        
+        # Create configuration file for coverity buildless
+        echo -e 'capture:\n   build-command-inference: false' > "$SOURCE_CODE_DIR"/coverity.yml
+        
+        set +e -x
+        # Buildless scan
+        env COV_HOST=konflux HOME=/var/tmp/coverity/home /opt/coverity/bin/coverity capture $COV_CAPTURE_ARGS --project-dir "$SOURCE_CODE_DIR" --dir "$COVERITY_DIR"
+        COV_CAPTURE_EXIT_CODE=$?
+        set -x
+        
+        
+        if [[ "$COV_CAPTURE_EXIT_CODE" -eq 0 ]]; then
+            echo "Coverity capture scan finished successfully"
+        else
+            echo "Coverity capture command failed with exit code ${COV_CAPTURE_EXIT_CODE}. Exiting..."
+            exit 1
+        fi
+        
+        # Analysis phase
+        set -x
+        /opt/coverity/bin/cov-manage-emit --dir $COVERITY_DIR reset-host-name
+        /opt/coverity/bin/cov-analyze $COV_ANALYZE_ARGS --dir="$COVERITY_DIR"
+        COV_ANALYZE_EXIT_CODE=$?
+        set +x
+        
+        if [[ "$COV_ANALYZE_EXIT_CODE" -eq 0 ]]; then
+          echo "Coverity analyze scan finished successfully"
+        else
+          echo "Coverity analyze scan failed with exit code ${COV_ANALYZE_EXIT_CODE}. Exiting..."
+          exit 1
+        fi
+        set -e
+        
+        /opt/coverity/bin/cov-format-errors --dir=$COVERITY_DIR --json-output-v10 "$COVERITY_RESULTS_FILE"
+        # We parse the results, embed context, remove duplicates and store them in SARIF format.
+        IMP_LEVEL=1
+        if [ "${IMP_FINDINGS_ONLY}" == "false" ]; then
+          IMP_LEVEL=0
+        fi
+        
+        (cd "$SOURCE_CODE_DIR" && csgrep --mode=json --imp-level="$IMP_LEVEL" --remove-duplicates --embed-context=3 "$COVERITY_RESULTS_FILE") \
+          | csgrep --mode=json --strip-path-prefix="$SOURCE_CODE_DIR"/source/ \
+          | csgrep --mode=json --strip-path-prefix="/var/tmp/coverity/home" \
```

Comment:
The `/var/tmp/coverity/home` path should ideally be set at a single place in the script in case we decide to change it later on.  If you export the variable at the beginning of the script, you do not need to explicitly set it while invoking `cov-build`.
```sh
export HOME="/var/tmp/coverity/home"
```
Then you can use `--strip-path-prefix="$HOME"` here.

Reply:
Replaced and using variable across thetask